### ISSUE TITLE
Toggling any module ON on Jetpack settings results in React warning

### ISSUE THREAD

**Issue by manzoorwanijk** [Automattic/jetpack#40086]
### Impacted plugin

Jetpack

### Quick summary

When a Jetpack module is OFF and you try to turn it ON on Jetpack settings page, it results in the warning:

```
A component is changing an uncontrolled input to be controlled.
```

![Image](https://github.com/user-attachments/assets/019b450c-47e1-43ba-b4fe-3c72bb4322bc)

### Steps to reproduce

- Goto Jetpack settings, e.g. `/wp-admin/admin.php?page=jetpack#/discussion`
- Enable a module, e.g. **Comments**
- Notice the warning in the browser console.

### A clear and concise description of what you expected to happen.

There should be no warning.

### What actually happened

A warning is shown in browser console

### Impact

All

### Available workarounds?

No and the platform is unusable

### Platform (Simple and/or Atomic)

Self-hosted, Atomic

### Logs or notes

The bug seems to have been there for [six years](https://github.com/Automattic/jetpack/pull/8934) and the source of that bug is the OR (`||`) operator here. It should instead be a null coalescing operator.

https://github.com/Automattic/jetpack/blob/5e41cdf798b5872dfed14c7387aac2a727c6657e/projects/plugins/jetpack/_inc/client/components/module-toggle/index.jsx#L99

```diff
- checked={ this.props.activated || this.props.isModuleActivated }
+ checked={ this.props.activated ?? this.props.isModuleActivated }
```

**Comment by anomiex** [comment]
I note that `??` will give different results for two cases. One is what you intend, but the other probably isn't. And one that probably should be changed isn't being changed either, although due to https://github.com/Automattic/jetpack/blob/5e41cdf798b5872dfed14c7387aac2a727c6657e/projects/plugins/jetpack/_inc/client/components/module-toggle/index.jsx#L24 that case may not be able to happen.

| `this.props.activated` | `this.props.isModuleActivated` | `\|\|` | `??` |
|--------|--------|--------|--------|
| _undefined_ | _undefined_ | _undefined_ | _undefined_ |
| undefined | true | true | true |
| undefined | false | false | false |
| true | undefined | true | true |
| true | true | true | true |
| true | false | true | true |
| _false_ | _undefined_ | _undefined_ | _false_ |
| _false_ | _true_ | _true_ | _false_ |
| false | false | false | false |

In other words, if `this.props.activated` is false while `this.props.isModuleActivated` is true, changing to `??` will change it from checked to unchecked.

If we don't want that change, using `this.props.activated || this.props.isModuleActivated || false`, or ensuring that `isModuleActivated` is not undefined, or if `activated` already can't be undefined swapping it to `this.props.isModuleActivated || this.props.activated` would work better.

Or another possibility is that `isModuleActivated` never seems to be used in the first place, and I wonder if #8934 had intended for it to be used as something like `this.props.isModuleActivated( this.props.slug )`. We could just drop it entirely.

**Comment by manzoorwanijk** [comment]
Yeah, I'm not sure about the logic here. `??` was just a thought.
Yeah, it will be good to get rid of that second condition.
